src/organization/organization.service.ts

```diff
@@ -1,19 +1,76 @@
 import { Injectable, Logger } from '@nestjs/common';
-import { OCTOKIT } from '../lib';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { RepositoryService } from '../repositories/repository.service';
+import { updateArray } from '../lib';
 import { User } from '../model';
+import { OrganizationDocument } from './model/schemas';
+import { queryMembers, queryRepos } from './lib';
 
 @Injectable()
 export class OrganizationService {
   private readonly logger = new Logger(OrganizationService.name);
 
-  public async orgaMembers(owner: string) {
-    const orgMembers: User[] = await OCTOKIT.rest.orgs
-      .listMembers({
-        org: owner,
-      })
-      .then((res) => res.data);
+  constructor(
+    @InjectModel('Organization')
+    private readonly orgModel: Model<OrganizationDocument>,
+    private repoService: RepositoryService,
+  ) {}
 
-    //TODO: create org schema. save all repos and org members of an org to that schema
+  public async initializeOrga(owner: string, repoNames?: string[]) {
+    this.logger.log(`creating org for ${owner}`);
+    const id = (await this.getOrg(owner))._id;
+    // TODO: this can be done in parallel and there should be no awaits
+    // necessary. will keep them for now for better testability and
+    // more stability in local runs.
+    await this.addRepos(owner, repoNames);
+    await this.addOrgaMembers(id, owner);
+  }
+
+  /**
+   * Creates org in database if it doesn't exist already.
+   * Else returns existing org.
+   */
+  private async getOrg(owner: string): Promise<OrganizationDocument> {
+    const exists = await this.orgModel.exists({
+      owner: owner,
+    });
+    if (exists) {
```

Comment:
I really dislike the `return await` pattern. There is no much different in function anyways since a promise is always returned. but in the case that the request fails the error handling is different